app/nasa_collaboration/page.js

```diff
@@ -1,72 +1,92 @@
 "use client";
 
 import React, { useState, useEffect } from 'react';
-import styles from './page.module.css';
+import styles from "./page.module.css"
 
-// Read "/app/nasa_collaboration/README.md" for more info about the API_KEY
-// You need a proper API_KEY for the requests to work
-const API_KEY = 'API_KEY';
+const API_KEY = 'TaA8BgllKVYPCxHa4t6SkF5NemO6QmTGv4lwvRkM';
 
 const NASA_URLs = {
-  astronomyPicOfTheDay: `https://api.nasa.gov/planetary/apod?api_key=${API_KEY}`,
+  astronomyPicOfTheDay:  `https://api.nasa.gov/planetary/apod?api_key=${API_KEY}`,
   marsRoverPhoto: `https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/photos?earth_date=2015-6-3&api_key=${API_KEY}`,
 };
 
+  const RoverPhoto = ({src, date, roverName}) => {
+  return (
+      <div className={styles.nasaPicOfTheDayImg}>
+          <img src={src} alt={`Rover: ${roverName} on ${date}`}/>
+          <p className={styles.roverText}>{`Date: ${date}`}</p>
+          <p className={styles.roverText}>{`Rover: ${roverName}`}</p>
+      </div>
+  );
+}
+ 
 export const NasaCollaboration = () => {
   const [dailyImg, setDailyImg] = useState({});
   const [roverPhoto, setRoverPhoto] = useState({});
 
   useEffect(() => {
     const fetchRoverPhotos = async () => {
-      const roverPhotoResponse = await fetch(NASA_URLs.marsRoverPhoto).then(response => response.json());
-      setRoverPhoto(roverPhotoResponse);
+      try {
+        const response = await fetch(NASA_URLs.marsRoverPhoto);
+        const data = await response.json();
+        setRoverPhoto(data);
+      } catch (error) {
+        console.error("Error fetching Mars rover photo:", error);
```

Comment:
How is this error communicated to the user? I.e., how will the user know something went wrong? :)

Reply:
The ui displays the error message now, thanks for the feedback.